src/server/node_manager.c

```diff
@@ -7119,6 +7119,24 @@ set_nodes(void *pobj, int objtype, char *execvnod_in, char **execvnod_out, char
 				return (PBSE_UNKNODE);
 			}
 
+                        if ((pnode->nd_state & (INUSE_DOWN | INUSE_STALE)) && (svr_init == FALSE)) {
+
+				if(objtype == JOB_OBJECT) {
+                                   free(phowl);
+                                   free(execvncopy);
+                                   return (PBSE_BAD_NODE_STATE);
+				} 
+				else {
+					if(presv->ri_qs.ri_state == RESV_UNCONFIRMED) {
+						set_resv_retry(presv, time_now+10);
```

Comment:
This is unneeded and has bad side effects.  This function was meant for reservations in the degraded state.  It sets the retry time.  That isn't supposed to be set until the resv is degraded.  The scheduler will take a rejection as a retry.  It will immediately stop its cycle and start a new one.  If a node has gone down or stale on the scheduler since it queried the world, the new cycle will see that.

Reply:
If the Reservation is still unconfirmed - is it Ok to be degraded with out nodes assigned to it and if we make it degraded, it will stay as degraded till node is available for Standing reservation. if the reservaion is unconfirmed even we make standing reservation is degraded it will not be rescheduled.

In case of normal reservation, even if it is degraded - it will not be considered for reschedule or retry resv in both the cases (unconfirmed or state moved from confirmed->degraded), 
even when nodes are available.
would you think directly using set_resv_retry call will impact scheduling cycle?

Standing Reservation:
if degraded, it will be put back to resv retry.
if unconfirmed, it will be rejected, so I've manually used set_resv_retry to put it back for retry.
In case of Standing resv, setting the main state to degraded will mean to put it for rescheduling/reallocating the nodes.

Normal Reservation:
if degraded, the nodes will not be reallocated with new ones.
if unconfirmed, the current changes will put the resv to try again (set_resv_retry), if the scheduled node is down/stale.
In case of normal reservation, setting the reservation state to degraded, it will not be set to resv retry.

Reply:
The issue is with calling set_resv_retry().  It does more than just sets a task on the list to retry the reservation.  It also sets the internal attribute/member for retrying.  If the reservation goes degraded later, these are supposed to be unset.  If you want to just kick the scheduler to retry, set a work task for it (look at set_resv_retry() for how to do it).  Now you do need to call set_resv_retry for when you set the reservation in the degraded state on the other side of the if.